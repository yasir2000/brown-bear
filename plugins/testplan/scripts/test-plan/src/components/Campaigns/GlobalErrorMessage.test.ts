/*
 * Copyright (c) Enalean, 2020 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import type { Wrapper } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import { createTestPlanLocalVue } from "../../helpers/local-vue-for-test";
import GlobalErrorMessage from "./GlobalErrorMessage.vue";
import { createStoreMock } from "../../../../../../../src/scripts/vue-components/store-wrapper-jest";
import type { RootState } from "../../store/type";
import type { CampaignState } from "../../store/campaign/type";

describe("GlobalErrorMessage", () => {
    async function createWrapper(campaign: CampaignState): Promise<Wrapper<GlobalErrorMessage>> {
        return shallowMount(GlobalErrorMessage, {
            localVue: await createTestPlanLocalVue(),
            mocks: {
                $store: createStoreMock({
                    state: {
                        campaign,
                    } as RootState,
                }),
            },
        });
    }

    it("displays nothing if there is no error", async () => {
        const wrapper = await createWrapper({
            has_refreshing_error: false,
        } as CampaignState);

        expect(wrapper.element).toMatchInlineSnapshot(`<!---->`);
    });

    it("displays error message when a campaign cannot be refreshed", async () => {
        const wrapper = await createWrapper({
            has_refreshing_error: true,
        } as CampaignState);

        expect(wrapper.element).toMatchSnapshot();
    });
});
