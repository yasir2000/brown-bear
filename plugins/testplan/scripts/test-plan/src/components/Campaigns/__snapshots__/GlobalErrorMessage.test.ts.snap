// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalErrorMessage displays error message when a campaign cannot be refreshed 1`] = `
<div
  class="tlp-alert-danger test-plan-list-of-campaigns-error"
  data-current-language="en_US"
  data-msgid="
    An error occurred while retrieving information of the new campaign.
"
>
  
    An error occurred while retrieving information of the new campaign.

</div>
`;
